rpc/block_svc_routines.c

```diff
@@ -4990,7 +4991,7 @@ block_list_cli_1_svc_st(blockListCli *blk, struct svc_req *rqstp)
   if (!tgmdfd) {
     errCode = errno;
     GB_ASPRINTF (&errMsg, "Not able to open metadata directory for volume "
-                 "%s[%s]", blk->volume, strerror(errCode));
+                 "%s[%s]", blk->volume, strerror(errno));
```

Comment:
GB_ASPRINTF macro can override errno set by glfs_opendir(). That is why we are using strerror(errCode);

Anything wrong here ? what is the coverity msg you saw ?

Reply:
the wrong thing is, errCode is of type int, and it is checked for < 0 few lines below. Which means, for coverity, if errCode is -ve, we shouldn't be passing the value to strerror().

errno getting lost is valid concern, and I guess I will solve only that here.